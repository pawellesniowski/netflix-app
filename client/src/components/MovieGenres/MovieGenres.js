import React from 'react';
import './MovieGenres.scss';

const MovieGenres = ({genres = []}) => {
  return (
    <div className="movie-release-day">
      {genres.map(gener => <span key={gener}> {gener} </span>)}
    </div>
  )
};

export default MovieGenres;
