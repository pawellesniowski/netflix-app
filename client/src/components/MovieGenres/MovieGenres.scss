@import "../../common/styles/colors";

.movie-release-day {
  color: $color-grey-tuna;
  font-size: 1.2rem;
}
