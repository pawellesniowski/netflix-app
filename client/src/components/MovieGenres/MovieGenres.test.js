import React from "react";
import { shallow } from 'enzyme';
import MovieGenres from "./MovieGenres";


describe("MovieGenres with props", () => {
  const genres = ["drama", "horror"];
  let component;
  beforeEach(() => {
    component = shallow(<MovieGenres genres={genres}/>);
  });

  it("Displays list of genres", () => {
    expect(component.find('.movie-genres')).toHaveLength(1);
  });

  it("List has correct number of items", () => {
    expect(component.find('.item')).toHaveLength(genres.length);
  });

  it("Match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

});

describe("MovieGenres with NO props", () => {
  let component;
  beforeEach(() => {
    component = shallow(<MovieGenres />);
  });

  it("Displays list of genres", () => {
    expect(component.find('.movie-genres')).toHaveLength(1);
  });

  it("List has correct number of items", () => {
    expect(component.find('.item')).toHaveLength(0);
  });

  it("Match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

});