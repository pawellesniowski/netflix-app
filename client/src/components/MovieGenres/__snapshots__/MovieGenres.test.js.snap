// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGenres with NO props Match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MovieGenres with props Match the snapshot 1`] = `ShallowWrapper {}`;
